arangod/Cluster/ClusterInfo.cpp

```diff
@@ -1737,18 +1756,56 @@ void ClusterInfo::loadPlan() {
         if (newPlan.contains(StaticStrings::SystemDatabase)) {
           auto planSlice = newPlan[StaticStrings::SystemDatabase]->slice();
           if (planSlice.isArray() && planSlice.length() == 1) {
             if (planSlice.at(0).isObject()) {
               auto entrySlice = planSlice.at(0);
-              auto path = std::vector<std::string>{
-                  "arango", "Plan", "Databases", StaticStrings::SystemDatabase,
-                  StaticStrings::Sharding};
-              if (entrySlice.hasKey(path) && entrySlice.get(path).isString()) {
-                systemDB->setSharding(entrySlice.get(path).copyString());
+              {
+                // Add sharding Attribute
+                auto path = std::vector<std::string>{
+                    "arango", "Plan", "Databases",
+                    StaticStrings::SystemDatabase, StaticStrings::Sharding};
+                if (auto value = entrySlice.get(path); value.isString()) {
+                  systemDB->setSharding(value.copyString());
+                }
+              }
+              {
+                // Add replication version
+                auto path =
+                    std::vector<std::string>{"arango", "Plan", "Databases",
+                                             StaticStrings::SystemDatabase,
+                                             StaticStrings::ReplicationVersion};
+                if (auto value = entrySlice.get(path); value.isString()) {
+                  auto res = replication::parseVersion(value.stringView());
+                  // Just care for valid Replication versions now
+                  if (res.ok() && res.get() != systemDB->replicationVersion()) {
+                    // We cannot change the replication version of the system
+                    // database The system database is created during startup,
+                    // using the default option There is a timeframe where new
+                    // collections are already created for this database, and
+                    // use the "wrong" replicationVersion. This crashes on the
+                    // way before this codepoint is reached and could repair the
+                    // system Database.
+                    LOG_TOPIC("50b83", FATAL, arangodb::Logger::STARTUP)
+                        << "Changed option "
+                           "'--database.default-replication-version' between "
+                           "startups. (Was "
+                        << value.stringView() << ", is now set to "
+                        << replication::versionToString(
+                               systemDB->replicationVersion())
+                        << "). This would break the _system database. If you "
+                           "want your _system database to use a different "
+                           "replication version you need to start with an "
+                           "empty cluster and restore data.";
+                    FATAL_ERROR_EXIT();
```

Comment:
That doesn't sound right to me: I think the default replication version must never affect existing databases, only ever newly created ones. If it does, I think it's essential that we fix that (regardless of this PR).